packages/block-library/src/search/index.php

```diff
@@ -47,47 +44,38 @@ function render_block_core_search( $attributes ) {
 	$border_color_classes = get_border_color_classes_for_block_core_search( $attributes );
 
 	$label_inner_html = empty( $attributes['label'] ) ? __( 'Search' ) : wp_kses_post( $attributes['label'] );
-
-	$label_markup = sprintf(
-		'<label for="%1$s" class="wp-block-search__label screen-reader-text">%2$s</label>',
-		esc_attr( $input_id ),
-		$label_inner_html
-	);
-	if ( $show_label && ! empty( $attributes['label'] ) ) {
-		$label_classes = array( 'wp-block-search__label' );
-		if ( ! empty( $typography_classes ) ) {
-			$label_classes[] = $typography_classes;
+	$label            = new WP_HTML_Tag_Processor( sprintf( '<label %1$s>%2$s</label>', $inline_styles['label'], $label_inner_html ) );
```

Comment:
we're missing an opportunity here to rely on the safety of the HTML API by sprintf-ing the label. what's the shape of `$inline_styles['label']`? do we expect it to be something like ` style="…"`

this is _okay_ because that content is safely sourced, but we could also set it dynamically.

```php
//                              012345678
// $inline_styles['label'] === ` style="…"`
$style = substr( $inline_styles['label'], 8, -1 );

$label = new WP_HTML_Tag_Processor( '<label>' );
$label->next_tag();
$label->set_attribute( 'style', $style );
$label->set_attribute( 'for', $input_id );
…

$label_markup = $label->get_updated_html() . $label_inner_html . '</label>'; 
```

a similar comment applies below. in general it would be great if we avoided merging safe and unsafe domains, that is, it would best to avoid using `sprintf` to add HTML syntax inside of the tag processor string if we can help it.

I'm curious if doing this causes any double-escaping issues since inline styles are already escaped. if that's the case we might need to see if we can account for that in the HTML API.

otherwise: nice work - I like the refactor